### PR TITLE
Folder restructure bm core

### PR DESCRIPTION
**Pull request by matt001k**
Restructures the folder to match agreed upon notion document from our discussions on Monday.
Also, updates/adds CMake files to adjust for new structure.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -79,40 +73,15 @@ if(ENABLE_TESTING)
...
-        src/bm_service.c
-        src/bm_service_request.c
         ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src/cborparser.c
         ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src/cborencoder_float.c
         ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src/cborencoder.c
         ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src/cborerrorstrings.c
         ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src/cborvalidation.c
-        ${BM_COMMON_MESSAGES_SOURCES}
+        ${CMAKE_CURRENT_LIST_DIR}/third_party/crc/crc16.c
+        ${CMAKE_CURRENT_LIST_DIR}/third_party/crc/crc32.c
```
Comment (matt001k) [inline CMakeLists.txt:84]: Do we want to create a 3rd party library?

Comment (towynlin) [reply CMakeLists.txt:82]: Per slack discussion, I'd vote no. The main value of building static libraries is to prove to ourselves that we've minimized the dependencies in our own code. We already know these 3rd party libraries don't depend on bm_core. 😄

File: docs/usage.md [hunk 2]
```diff
@@ -6,6 +6,14 @@ with your own firmware,
 and you're wondering what would be required to make your device compatible with Bristlemouth,
 read on!
 
+## Setting Up The Repository
+The repository can be added as a git submodule to a project utilizing the `git submodule add` command,
+see [this](https://git-scm.com/book/en/v2/Git-Tools-Submodules) for more information.
+Or placed into a project by any other method.
+If cloned/added as a submodule to your project,
+make sure you update all of the submodules in `bm_core` by  `cd`'ing into `bm_core` and running:
+`git submodule update --init`.
```
Comment (towynlin) [inline docs/usage.md:15]: 🙏 👍

**Review by towynlin — APPROVED** [review]
Looks great! Thanks! Wooo moving lots of files!

I called out one place where you changed something back to the original way I did it that at the time didn't work. If it works now, awesome!

File: CMakeLists.txt [hunk 3]
```diff
@@ -59,16 +51,17 @@ set(COMPILE_FLAGS
...
+    ${CMAKE_CURRENT_LIST_DIR}/common
+    ${CMAKE_CURRENT_LIST_DIR}/middleware
+    ${CMAKE_CURRENT_LIST_DIR}/network
+    ${CMAKE_CURRENT_LIST_DIR}/third_party
+    ${CMAKE_CURRENT_LIST_DIR}/third_party/crc
     ${CMAKE_CURRENT_LIST_DIR}/third_party/tinycbor/src
     bm_common_messages
 )
-
-add_subdirectory("drivers/adin2111")
```
[the remark this answers is not in the document]
Comment (matt001k) [reply CMakeLists.txt:71]: Yep! They do, that makes sense of the removal of the static linking although.

[the remark this answers is not in the document]
Comment (towynlin) [reply CMakeLists.txt:71]: And I see now that the last few commits on this PR got the BM_INCLUDES set up correctly and moved out to the top outside the if-else block, which might have been related to whatever problems I was having before as well. 👍

**Review by victorsowa12 — APPROVED** [review]

**Pull request merged**